tests/perf/allocator_perf.cc

```diff
@@ -0,0 +1,139 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (C) 2023 ScyllaDB
+ */
+
+#include <seastar/testing/perf_tests.hh>
+
+#include <seastar/core/memory.hh>
+#include <sys/mman.h>
+
+struct allocator_test {
+    static constexpr size_t COUNT = 1000;
+    std::array<void *, COUNT> _pointers{};
+};
+
+struct allocator_test_split : public allocator_test {
+    enum alloc_test_flags {
+      MEASURE_ALLOC = 1 << 0,
+      MEASURE_FREE = 2 << 0,
+    };
```

Comment:
please use 4-space indents throughout

Reply:
also 1 << 0, 2 << 0? Did you mean 1<< 0, 1 << 1?

Reply:
Let's say this was an easter egg to ensure reviewers were awake which will be removed on a subsequent push :).

Reply:
This is addressed (fix and offending commit folded) in https://github.com/scylladb/seastar/commit/9d5d2445547b0eb47feef73a708dc9aeb1fbe005.

Reply:
oh come on, the only thing that triggers me more is trailing whitespace

Reply:
This reminded me to dust off my .clang-format patch for seastar. Get ready for some vigorous debate!